cluster-autoscaler/simulator/clustersnapshot/predicate/predicate_snapshot.go

```diff
@@ -32,11 +32,44 @@ type PredicateSnapshot struct {
 
 // NewPredicateSnapshot builds a PredicateSnapshot.
 func NewPredicateSnapshot(snapshotStore clustersnapshot.ClusterSnapshotStore, fwHandle *framework.Handle, draEnabled bool) *PredicateSnapshot {
-	return &PredicateSnapshot{
+	snapshot := &PredicateSnapshot{
 		ClusterSnapshotStore: snapshotStore,
-		pluginRunner:         NewSchedulerPluginRunner(fwHandle, snapshotStore),
 		draEnabled:           draEnabled,
 	}
+	snapshot.pluginRunner = NewSchedulerPluginRunner(fwHandle, snapshot)
```

Comment:
That hurts, but okay.

Reply:
If golang is a flat earth, I feel like we've fallen off the edge

Reply:
This one isn't actually a Golang issue, more of a code structure one. I added a comment explaining why it's there and left a TODO for improving it (I think a simple refactor would remove the `PluginRunner` dependency on `PredicateSnapshot`). In the interest of time and the 1.32 release I'd prefer not to tackle this one in this PR.

Reply:
Right, I think our object composition has resulted in this internationally recognized metaphor:

<img width="179" alt="image" src="https://github.com/user-attachments/assets/6fddd40b-ff98-40ac-8ced-d4cf1a7f9b88">